classes/local/report/tag_count_report_builder.php

```diff
@@ -0,0 +1,51 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_objectfs\local\report;
+
+/**
+ * Tag count report builder.
+ *
+ * @package   tool_objectfs
+ * @author    Matthew Hilton <matthewhilton@catalyst-au.net>
+ * @copyright Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class tag_count_report_builder extends objectfs_report_builder {
+    /**
+     * Builds report
+     * @param int $reportid
+     * @return objectfs_report
+     */
+    public function build_report($reportid) {
+        global $DB;
+        $report = new objectfs_report('tag_count', $reportid);
+
+        // Returns counts + sizes of key:value.
+        $sql = "
+            SELECT CONCAT(COALESCE(object_tags.tagkey, '(untagged)'), ': ', COALESCE(object_tags.tagvalue, '')) as datakey,
+                   COUNT(objects.id) as objectcount,
+                   SUM(objects.filesize) as objectsum
+              FROM {tool_objectfs_objects} objects
+         LEFT JOIN {tool_objectfs_object_tags} object_tags
```

Comment:
I'm quite anxious about this left join at scale

Reply:
This is in the report builder so it will be done via adhoc tasks and never hit via web. However saying that the reason its joining is to get the filesize which is currently expected by the objectfs report handlers, despite it not really being that useful in the context of the report.
I could probably tweak that API a bit so we don't need the filesize hence dont need the join will see what I can do

Reply:
Turns out it was really simple to remove the object size from the report and hence remove the need for the left join - have changed this in https://github.com/catalyst/moodle-tool_objectfs/pull/619/commits/3002355ed9c5e46669fee456416daf5b718ae0ff